src/developer/core/plugin-api.md

```diff
@@ -69,16 +72,159 @@ This is the workhorse.
+plugins via callbacks (see below). These are defined in the root
+[`Plugin` class](https://github.com/pupil-labs/pupil/blob/master/pupil_src/shared_modules/plugin.py).
+In order to create a new plugin, one has to inherit from this class and overwrite them
+as desired.
+
 ```py
 from plugin import Plugin
 
 class MyCustomPlugin(Plugin):
     pass
 ```
 
+Plugins are automatically listed in the [Plugin Manager](/core/software/pupil-capture/#plugins)
+unless they inherit from a set of special classes (`System_Plugin_Base`, `Base_Manager`,
+`Base_Source`, `CalibrationChoreographyPlugin`, `GazerBase`).
+
+### Plugin class attributes
+
+These class attributes define general plugin behavior. Overwriting them is optional but
+recommended. `alive` is an exception and should only be set to `False` if you want the
+plugin to close autonomously. Otherwise, this attribute is managed by the enclosing
+application.
+
+| Name         | Possible values                                                                                        | Default value | Meaning                                                              |
+|--------------|--------------------------------------------------------------------------------------------------------|---------------|----------------------------------------------------------------------|
+| `uniqueness` | `"not_unique"`, `"by_class"`, `"by_base_class"`                                                        | `"by_class"`  | Plugin instance replacement behavior. See below.                     |
+| `order`      | float in the range of [0.0, 1.0]                                                                       | `0.5`         | Defines the order in which plugins are loaded and called             |
+| `icon_font`  | `"roboto"`, `"pupil_icons"`, `"opensans"`, any other font registered via `Plugin.g_pool.ui.add_font()` | `"roboto"`    | Menu icon font                                                       |
+| `icon_chr`   | Any string whose letters are present in `icon_font`. Recommended to use a single letter string.        | `"?"`         | Menu icon                                                            |
+| `alive`      | `True`, `False`                                                                                        | `True`        | Setting to `False` will shutdown the plugin in the next event cycle. |
+
+
+#### Plugin uniqueness
+- *not unique* - plugins can be instantiated multiple times, e.g. gaze visualization
+plugins
+- *unique by class* - only one plugin instance can exist at a time, e.g. blink
+detector
+- *unique by base class* - if two  plugins share the same base class they are only
+allowed to be active one at a time. For example, each calibration choreography is
+implemented as a separate *by-base-class*-unique plugin but since they all share the
+same base class (`CalibrationChoreographyPlugin`) only one can be active at a time.
```

Comment:
I like the suggestion!
```suggestion
- *unique by base class* - if two  plugins share the same base class they are only
allowed to be active one at a time. Calibration choreographies are examples of *unique by base class* plugins.
They each implement a separate *by-base-class*-unique plugin but since they all share the
same base class (`CalibrationChoreographyPlugin`) only one can be active at a time.
```